runtime/j9vm/jvm.c

```diff
@@ -1514,6 +1524,88 @@ static jint initializeReflectionGlobals(JNIEnv * env, BOOLEAN includeAccessors)
  *	DLL: jvm
  */
 jint JNICALL JNI_CreateJavaVM(JavaVM **pvm, void **penv, void *vm_args) {
+	return JNI_CreateJavaVM_impl(pvm, penv, vm_args, FALSE);
+}
+
+#if defined(JITSERVER_SUPPORT)
+int32_t JNICALL 
+JITServer_CreateServer(JITServer **jitServer, void *vm_args)
+{
+	JNIEnv *env = NULL;
+	JITServer *server = NULL;
+	jint rc = JITSERVER_OK;
+
+	server = malloc(sizeof(JITServer));
+	if (NULL == server) {
+		rc = JITSERVER_OOM;
+		goto _end;
+	}
+	server->startJITServer = startJITServer;
+	server->waitForJITServerTermination = waitForJITServerTermination;
+	rc = JNI_CreateJavaVM_impl(&server->jvm, (void **)&env, vm_args, TRUE);
+
+	if (JNI_OK != rc) {
+		rc = JITSERVER_CREATE_FAILED;
+		goto _end;
+	}
+	*jitServer = server;
+	rc = JITSERVER_OK; /* success */
+
+_end:
+	if ((JITSERVER_OK != rc) && (NULL != server)) { 
+		free(server);
+	}
+	return rc;
+}
+
+static int32_t 
+startJITServer(JITServer *jitServer)
+{
+	JavaVM *vm = jitServer->jvm;
+	JNIEnv *env = NULL;
+	int32_t rc = JITSERVER_OK;
+
+	rc = (*vm)->AttachCurrentThread(vm, (void **)&env, NULL);
+	if (JNI_OK == rc) {
+		J9JITConfig *jitConfig = ((J9JavaVM *)vm)->jitConfig;
+		rc = jitConfig->startJITServer(jitConfig);
```

Comment:
`startJITServer` returns 0 on  success, -1 on error. I think its easier to manage if JIT return/error codes are kept separate from the JITServer interface return/error codes.

Reply:
It means we potentially lose information here - there is no path to find out why the startJITServer call failed

Reply:
I agree with that, but I think we should be adding tracing for this instead of trying to cover each and every failure that can happen.
Moreover, the Listener thread is not currently passing any error information down the stack.